### PR TITLE
throw exception if policy does not exist

### PR DESCRIPTION
**Pull request by martinfurmanski**
The plugins can now throw procedure exceptions and the server policies
plugin will throw a ProcedureCallFailed if the supplied policy name
does not exist.

### CODE REVIEW

File: enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/load_balancing/plugins/server_policies/Policies.java [hunk 1]
```diff
@@ -31,7 +33,7 @@
 {
     public static final String POLICY_KEY = "load_balancing.policy"; // TODO: move somewhere (driver support package?)
```
Comment (zhenlineo) [inline enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/load_balancing/plugins/server_policies/Policies.java:34]: Can we change this key to something else?
One plan is to pass routing context via bolt url, in we go down that road, then it is really a bad idea to have a `.` in the parameters as it requires a skip in url.

I would vote for `policy` or `load_balancing_policy`.

**Review by zhenlineo — CHANGES_REQUESTED** [review]

**Comment by digitalstain** [comment]
Just a note that this also allows for overriding the default default policy, which is a change in behaviour in addition to the exception being thrown.
Looks good to me.

**Comment by martinfurmanski** [comment]
@digitalstain That is not accurate I think! What you describe is captured by shouldAllowOverridingDefaultPolicy and that has not changed. There is a change in behaviour here however, expressed in the change of other tests!

**Comment by digitalstain** [comment]
@martinfurmanski Correct, my bad. Still looks good to me.

**Pull request merged**